kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/system/SystemAdminServiceImpl.java

```diff
@@ -128,42 +129,33 @@ public String getUptime() {
                 }
             }
         } else if (OS_MAC_OSX.equals(getOsName())) {
+        	Process p = null;
+        	BufferedReader br = null;
+        	
             try {
-                String systemUptime = runSystemCommand("uptime", false, this.executorService);
-                if (!systemUptime.isEmpty()) {
-                    String[] uptimeParts = systemUptime.split("up\\s+")[1].split("\\s*,\\s*");
-                    int days = 0, hours = 0, mins = 0;
-
-                    String uptimePart = uptimeParts[0];
-
-                    // If up less than a day, it will only show the number of mins, hr, or HH:MM
-                    if (uptimePart.contains("days")) {
-                        days = Integer.parseInt(uptimePart.split("\\s+days")[0]);
-                        uptimePart = uptimeParts[1];
-                    } else if (uptimePart.contains("day")) {
-                        days = Integer.parseInt(uptimePart.split("\\s+day")[0]);
-                        uptimePart = uptimeParts[1];
-                    }
-
-                    if (uptimePart.contains(":")) {
-                        // Showing HH:MM
-                        hours = Integer.parseInt(uptimePart.split(":")[0]);
-                        mins = Integer.parseInt(uptimePart.split(":")[1]);
-                    } else if (uptimePart.contains("hr")) {
-                        // Only showing hr
-                        hours = Integer.parseInt(uptimePart.split("\\s*hr")[0]);
-                    } else if (uptimePart.contains("mins")) {
-                        // Only showing mins
-                        mins = Integer.parseInt(uptimePart.split("\\s*mins")[0]);
-                    } else {
-                        logger.error("uptime could not be parsed correctly: {}", uptimeParts[0]);
-                    }
-
-                    uptime = (long) ((days * 24 + hours) * 60 + mins) * 60;
-                    uptimeStr = Long.toString(uptime * 1000);
-                }
+            	p = Runtime.getRuntime().exec("sysctl -n kern.boottime");
```

Comment:
Why do you use the `Runtime.exec` here? The class is already bound to the `CommandExecutorService` that was developed to run command on the system. I suggest to replace it with the `runSystemCommand` method that internally uses the `CommandExecutorService` and return a string representing the output of the command.

Reply:
I see your point. Changed to use CommandExecutorService to run system command.